code/lighting/lighting_profiles.cpp

```diff
@@ -154,6 +154,11 @@ piecewise_power_curve_intermediates lighting_profile::current_piecewise_intermed
 piecewise_power_curve_intermediates lighting_profile::calc_intermediates(piecewise_power_curve_values input){
 
 	piecewise_power_curve_intermediates ppci;
+	CLAMP(input.toe_length, 0.0f, 1.0f);
+	CLAMP(input.toe_strength, 0.0f, 1.0f);
+	CLAMP(input.shoulder_angle, 0.0f, 1.0f);
+	CLAMP(input.shoulder_length, 0.0f, 1.0f);
+	input.shoulder_strength = fmax(0.0f,input.shoulder_strength);
```

Comment:
nitpick, but having a space after the comma to match the preceding style would be good

Reply:
Also, might be worthwhile to add a comment mentioning `these clamps adds some safety boundaries from John Hable's implementation` or something to that affect

Reply:
done and done